chain/cosmos/chain_node.go

```diff
@@ -427,82 +423,6 @@ func (tn *ChainNode) Height(ctx context.Context) (int64, error) {
 	return height, nil
 }
 
-// FindTxs implements blockdb.BlockSaver.
-func (tn *ChainNode) FindTxs(ctx context.Context, height int64) ([]blockdb.Tx, error) {
```

Comment:
refactor this to query cometbft for a given block, get the Txs, base64 decode, and query each tx hash from consensus directly to get the EventAttributes. Then query the block to get the FinalizeBlockEvents.

While this is less efficient, it removes a TON of code for a function that does not seem to be called hardly ever